src/utils/calculate-max-and-min-date.ts

```diff
@@ -0,0 +1,15 @@
+import { TcDate } from '@timechimp/timechimp-typescript-helpers';
+
+export const calculateMaxAndMinDate = (date: string, maxDate: Date | undefined, minDate: Date | undefined) => {
```

Comment:
@Ihor-Lavrov maybe it's better to do `maxDate?: Date` instead of `maxDate: Date | undefined` ?

Reply:
Undefined is possible value here during initial call